Source/WebKit/Shared/ShareableBitmap.h

```diff
@@ -35,23 +35,58 @@
 #include <wtf/ThreadSafeRefCounted.h>
 
 namespace WebCore {
-class Image;
 class GraphicsContext;
+class Image;
+class NativeImage;
 }
 
 namespace WebKit {
 
-struct ShareableBitmapConfiguration {
-    std::optional<WebCore::DestinationColorSpace> colorSpace;
-    bool isOpaque { false };
+class ShareableBitmapConfiguration {
+public:
+    ShareableBitmapConfiguration() = default;
+
+    ShareableBitmapConfiguration(const WebCore::IntSize&, std::optional<WebCore::DestinationColorSpace> = std::nullopt, bool isOpaque = false);
+    ShareableBitmapConfiguration(const WebCore::IntSize&, std::optional<WebCore::DestinationColorSpace>, bool isOpaque, unsigned bytesPerPixel, unsigned bytesPerRow, unsigned bitmapInfo);
+
+    WebCore::IntSize size() const { return m_size; }
+    const WebCore::DestinationColorSpace& colorSpace() const { return m_colorSpace ? *m_colorSpace : WebCore::DestinationColorSpace::SRGB(); }
+    WebCore::PlatformColorSpaceValue platformColorSpace() const { return colorSpace().platformColorSpace(); }
+    bool isOpaque() const { return m_isOpaque; }
+
+    unsigned bytesPerPixel() const { ASSERT(!m_bytesPerPixel.hasOverflowed()); return m_bytesPerPixel; }
+    unsigned bytesPerRow() const { ASSERT(!m_bytesPerRow.hasOverflowed()); return m_bytesPerRow; }
+    unsigned bitmapInfo() const { return m_bitmapInfo; }
+
+    CheckedUint32 sizeInBytes() const { return m_bytesPerRow * m_size.height(); }
+
+    static CheckedUint32 calculateBytesPerRow(const WebCore::IntSize&, const WebCore::DestinationColorSpace&);
+    static CheckedUint32 calculateSizeInBytes(const WebCore::IntSize&, const WebCore::DestinationColorSpace&);
+
+private:
+    friend struct IPC::ArgumentCoder<ShareableBitmapConfiguration, void>;
+
+    static std::optional<WebCore::DestinationColorSpace> validateColorSpace(std::optional<WebCore::DestinationColorSpace>);
+    static CheckedUint32 calculateBytesPerPixel(const WebCore::DestinationColorSpace&);
+#if USE(CG)
+    static unsigned calculateBitmapInfo(const WebCore::DestinationColorSpace&, bool isOpaque);
+#endif
+
+    WebCore::IntSize m_size;
+    std::optional<WebCore::DestinationColorSpace> m_colorSpace;
+    bool m_isOpaque { false };
+
+    CheckedUint32 m_bytesPerPixel;
+    CheckedUint32 m_bytesPerRow;
+    unsigned m_bitmapInfo { 0 };
```

Comment:
I'm not really a fan of using `unsigned` for bitmapInfo here in cross-platform code. It's bad enough that `bitmapInfo` is so weakly typed in CG code.